crates/iota-data-ingestion-core/src/reducer.rs

```diff
@@ -41,7 +43,7 @@ pub trait Reducer<Mapper: Worker>: Send + Sync {
     /// # Note
     /// Messages within each batch are guaranteed to be ordered by checkpoint
     /// sequence number.
-    async fn commit(&self, batch: Vec<Mapper::Message>) -> Result<(), Mapper::Error>;
+    async fn commit(&self, batch: Arc<Vec<Mapper::Message>>) -> Result<(), Mapper::Error>;
```

Comment:
Since you don't want to consume why not accepting `batch: &[Mapper::Message]`?

Reply:
fixed: https://github.com/iotaledger/iota/pull/5820/commits/a36cb9392bd604fbd8669ecf6d81fa044ef1257c